orangecontrib/text/import_documents.py

```diff
@@ -0,0 +1,262 @@
+            if ext in eval(reader).ext:
+                return eval("{}(path)".format(reader))
+        return Reader(path)
+
+    def read(self, ):
+        error = ""
+        try:
+            self.read_file()
+        except Exception as ex:
+            textdata = None
+            error = "{}: {}".format(pathlib.Path(self.path).name, str(ex))
+            log.exception('Error reading failed', exc_info=ex)
+        else:
+            textdata = self.make_text_data()
+        return textdata, error
+
+    def read_file(self):
+        raise NotImplementedError("No reader for {}".format(pathlib.Path(self.path).suffix))
+
+    def make_text_data(self):
+        name = pathlib.Path(self.path).stem
+        directory = pathlib.PurePath(self.path).parent
+        category = directory.parts[-1] or "None"
+        if self.replace_white_space:
+            self.content = re.sub('\s+', ' ', self.content)
+        return TextData(name, self.path, self.ext, category, self.content)
+
+
+class TxtReader(Reader):
+    ext = [".txt"]
+
+    def read_file(self):
+        encoding = detect_encoding(self.path)
+        with open(self.path, 'r', encoding=encoding) as f:
+            self.content = f.read()
+
+
+class DocxReader(Reader):
+    ext = [".docx"]
+
+    def read_file(self):
+        self.content = docx2txt.process(self.path)
+
+
+class OdtReader(Reader):
+    ext = [".odt"]
+
+    def read_file(self):
+        odtfile = load(self.path)
+        texts = odtfile.getElementsByType(text.P)
+        self.content = " ".join(teletype.extractText(t) for t in texts)
+
+
+class PdfReader(Reader):
+    """
+    char_margin — two text chunks whose distance is closer than this value are considered
+    contiguous and get grouped into one.
+    word_margin — it may be required to insert blank characters (spaces) as necessary if
+    the distance between two words is greater than this value, as a blank between words might
+    not be represented as a space, but indicated by the positioning of each word.
+    """
+    ext = [".pdf"]
+
+    def read_file(self):
+        with open(self.path, 'rb') as f:
+            parser = PDFParser(f)
+        doc = PDFDocument()
+        parser.set_document(doc)
+        doc.set_parser(parser)
+        doc.initialize('')
+        rsrcmgr = PDFResourceManager()
+        laparams = LAParams()
+        laparams.char_margin = 0.1
+        laparams.word_margin = 1.0
+        device = PDFPageAggregator(rsrcmgr, laparams=laparams)
+        interpreter = PDFPageInterpreter(rsrcmgr, device)
+        extracted_text = []
+
+        for page in doc.get_pages():
+            interpreter.process_page(page)
+            layout = device.get_result()
+            for lt_obj in layout:
+                if isinstance(lt_obj, LTTextBox) or isinstance(lt_obj, LTTextLine):
+                    extracted_text.append(lt_obj.get_text())
+        self.content = ' '.join(extracted_text)
+
+
+class XmlReader(Reader):
+    ext = [".xml"]
```

Comment:
I had some issues and `html` is not so high priority at the moment. But no worries, I am planning to implement other formats as well.